test/luatest_helpers/server.lua

```diff
@@ -53,6 +56,13 @@ function Server:initialize()
         fio.rmtree(self.workdir)
         fio.mktree(self.workdir)
     end
+    if self.copydir ~= nil then
+        local ok, err = fio.copytree(self.copydir, self.workdir)
+        if not ok then
+            error(string.format('Failed to copy directory: %s', err))
+        end
+        self.copydir = nil
+    end
```

Comment:
It looks like this could be used instead of

https://github.com/tarantool/tarantool/blob/2e9cbec3091e7c5d6ca1d7dad4305333e70f5876/test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua#L6-L10

Can you please patch this test in the scope of this commit/PR to prove your code is viable?

Reply:
Added follow-up commit with refactoring of this test.